src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
+  :- {:outcome (schema/enum :success :not-found :error) :message schema/Str}
+  "Delete pending certificate requests for subject"
+  [subject :- schema/Str
+   {:keys [csrdir] :as settings} :- CaSettings]
+  (let [csr-path (path-to-cert-request csrdir subject)]
+    (if (fs/exists? csr-path)
+      (if (fs/delete csr-path)
+        (let [msg (str "Deleted certificate request for " subject)]
+          (log/debug msg)
+          {:outcome :success
+           :message msg})
+        (let [msg (format "Path %s exists but could not be deleted" csr-path)]
```

Comment:
It seems fine to me, especially since it's just a path to a CSR.  If an attacker has root access on the master I doubt that knowing a path to a CSR that couldn't be deleted makes things any worse for us.